torch/_refs/__init__.py

```diff
@@ -458,7 +490,7 @@ def __maybe_broadcast(x, shape):
             if preserve_cpu_scalar_tensors and utils.is_cpu_scalar_tensor(x):
                 return x
 
-            if not utils.same_shape(x.shape, common_shape):
+            if should_expand(x.shape, common_shape):
```

Comment:
It seems like the old code would expand if it couldn't prove the shapes were the same but the new code assumes that unbacked are equal and thus doesn't expand w/ a runtime check. Is that the right thing to do? Or should we just always expand unbacked?
(The old code was "weird", right? Like "u0 vs u1" would not expand but "s0/constant vs u1" WOULD expand?)
i.e. Should the new code really do something like add "handle_broadcast" to same_shape() and then "if not same_shape(x.shape, common_shape, broadcast=True)" then expand?

Reply:
the old code assumes that trying to handle u0 ==1 in case in which we would want to expand) guard size oblivious will say well assume u0>2 hence expand
the new code preserve that handling for u0==1 but also address u0==u1 in this case what shall we do?
previously  guard size oblivious woudl throw DDE, in this new code we will assume u0 and u1 equal.

I am not sure if for this function specifically we would receive that, but this is generally how we handle broadcasting case. (I hope we do that right in all locations now lol) /